rm: linux: development-tags: add new tags (#539)
foundriesio/docs: source/reference-manual/linux/development-tags.rst

The most common used tags through the Foundries.io repositories are:

* ``[FIO fromtree]``: patches cherry-picked, rather than merged, from upstream (mainline)

> * ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, peding PRs etc.), and revisions to them

Reviewer: ```suggestion
* ``[FIO fromlist]``: patches submitted to upstream for review (in mailing lists, pending PRs etc.), and revisions to them
```

> * ``[FIO toup]``: patches that want to go upstream/or have not been merged yet

Reviewer 2: `have not been upstreamed yet`. Patches `have not been merged` - it is `FIO fromlist` :)

> * ``[FIO temphack]`` or ``[FIO hack]``: temporarily patches that keep things working for now, but need a better solution later for upstreaming

Reviewer: ```suggestion
* ``[FIO temphack]`` or ``[FIO hack]``: temporary patches that keep things working for now but need a better solution later for upstreaming
```

* ``[FIO extras]``: patches pulled in for specific functionality that are useful for subscribers, but not critical
* ``[FIO internal]``: patches needed by the LmP, not intended for upstream use

> * ``[FIO squash]``: patch should be squashed with an original patch, fixing possible issues in that patch

Reviewer 2: I would add there something like: `This tag requires a commit message tag "Fixes:" filled out properly.`

> There are also exceptional tags for patches, that were cherry-picked/sent from/to SoC vendor forks:

Reviewer: ```suggestion
There are also exceptional tags for patches that were cherry-picked/sent from/to SoC vendor forks:
```

* ``[FIO from<vendor_name>]``: patches cherry-picked from ``<vendor_name>`` forked tree, for example ``[FIO fromnxp]``
* ``[FIO to<vendor_name>]``: patches, that want to go to SoC vendor forked tree, for example ``[FIO tostm]``

> * ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``

Reviewer: ```suggestion
* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts or it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
```

> * ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely re-implemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``

Reviewer 3: ```suggestion
* ``[FIO to<vendor_name>-altered]``: exceptional case, when not a whole patch was cherry-picked, but rather some parts of it or a it was completely reimplemented. In this case ``-altered`` is added, for example ``[FIO fromnxp-altered]``
```

> .. important:: When a patch is cherry-picked, git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so

Reviewer 3: ```suggestion
.. important:: When a patch is cherry-picked, the Git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
```

> .. important:: When a patch is cherry-picked, git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
>    original commit hash is added to the new commit message (for example ``(cherry-picked from commit 1e24c2671acdbcf81207c43da39e09846f404dc3)``).
>    With a hash it is much easier to track original commit a mainline/SoC vendor tree.

Reviewer: ```suggestion
.. important:: When a patch is cherry-picked, git cherry-pick command should be invoked with ``-x`` (append commit name) parameter, so
   the original commit hash is added to the new commit message (for example ``(cherry-picked from commit 1e24c2671acdbcf81207c43da39e09846f404dc3)``).
   With a hash, tracking the original commit in a mainline/SoC vendor tree is much easier.
```
